doc/metadata.md

```diff
@@ -72,22 +69,80 @@
         chunk chunks[];
     }
 
+    // this must be a fixed size so we can binary search over it easily
     struct inode {
         u64 ino;
         inode_type type;
+        u32 uid;
+        u32 gid;
+        u16 mode;
+        u64 mtime, atime, ctime; /* seems like we should require these? */
         union {
-            // directory
+            // fifo
+            struct {
+                /* nothing additional about fifos */
+            },
+
+            // chr
+            struct {
+                dev_t major;
+                dev_t minor;
+            },
+
+            // dir
             struct {
-                u64 offset;
+                u64 dir_offset;
```

Comment:
is this the offset to the `dir_list` struct?

Reply:
Yep, exactly.